### PR TITLE
immutable-variables-style

### LINKED ISSUE #12675: Immutable style guide
## Page

https://docs.soliditylang.org/en/v0.8.10/style-guide.html

## Abstract

There is nothing that references the immutable variable style guides. 

Should it follow the syntax of `CONSTANTS_WITH_UNDERSCORES` or `camelCase`?

### PR DESCRIPTION
**Pull request by NunoFilipeSantos**
Fixes #12675 .

### CODE REVIEW

File: docs/style-guide.rst [hunk 1]
```diff
@@ -1275,6 +1275,10 @@ Enums
 
 Enums, in the style of simple type declarations, should be named using the CapWords style. Examples: ``TokenGroup``, ``Frame``, ``HashStyle``, ``CharacterLocation``.
 
+Immutable variables
+===================
+
+
```
Comment (NunoFilipeSantos) [inline docs/style-guide.rst:1281]: @ekpyron and @cameel is it CONSTANTS_WITH_UNDERSCORES or camelCase?

Comment (cameel) [reply docs/style-guide.rst:1281]: Good question :) Ether one has something going for it.

On one hand, I think that the choice between a state variable and an immutable is sometimes just an implementation detail. So a convention matching state variables would make sense. On the other hand, there are important semantic differences between the two and some people may want immutables to stand out precisely to make it clear which one you're dealing with.

I have no personal opinion on that matter.

Looks like [solhint does not have a style rule for immutables either](https://github.com/protofire/solhint/blob/master/docs/rules.md#style-guide-rules). Only for constants.

Looking at examples from our external tests, people seem to be using both, though the style matching state variables seems to be more common.

Comment (NunoFilipeSantos) [reply docs/style-guide.rst:1281]: In terms of consistency with other parts of Solidity, what would be your guess between one or the other?

Comment (cameel) [reply docs/style-guide.rst:1281]: I wouldn't call either of them inconsistent. Both follow existing features and immutables have some features of both constants and state variables.

I'm not sure if making a choice about it now is a good idea. I'd be inclined to just mention both possibilities and wait until someone complains and backs it up with good arguments to choose one over the other.

Comment (NunoFilipeSantos) [reply docs/style-guide.rst:1281]: Thank you Kamil!
@ekpyron any opinion here?

Comment (ekpyron) [reply docs/style-guide.rst:1281]: Not really :-). Probably best to just search github for the most common practice and just document that. It'd make sense to align it with constants or with state variables or to make it deliberately distinct from either of them.

Comment (NunoFilipeSantos) [reply docs/style-guide.rst:1281]: 3 ORs. 😆 Cool, closing this.

**Pull request closed without merging**